pandas/core/groupby.py

```diff
@@ -1329,6 +1329,27 @@ def cumsum(self, axis=0):
 
     @Substitution(name='groupby')
     @Appender(_doc_template)
+    def rank(self, axis=0, method='average', numeric_only=True,
+             na_option='keep', ascending=True, pct=False):
+        """Compute numerical data ranks (1 through n) along axis.
+        """
+
+        if numeric_only:
+            data = self._obj_with_exclusions._get_numeric_data()
+            if data.size == 0:
+                raise DataError('No numeric types to aggregate')
+            data = data.groupby(self.grouper)
+        else:
```

Comment:
exactly what I said. The data is already selected by the time this is called (its called on each block). So the ``numeric_only`` flag is basically useless for this. you can *skip* things, but you are just creating more complicated code.